skglm/tests/test_estimators.py

```diff
@@ -191,8 +192,8 @@ def assert_deep_dict_equal(expected_attr, estimator):
                 np.testing.assert_allclose(v, v_est)
             else:
                 assert v == v_est
-    df_reg, df_clf = Quadratic(), Logistic()
-    pen_reg, pen_clf = L1(4.), MCPenalty(2., 3.)
+    df_reg, df_clf = compiled_clone(Quadratic()), compiled_clone(Logistic())
+    pen_reg, pen_clf = compiled_clone(L1(4.)), compiled_clone(MCPenalty(2., 3.))
```

Comment:
It's regular instances which should be passed rather than compiled ones, no?

Reply:
They are not compiled hence don't have a Lipschitz attribute.

Reply:
Compilation is not related to having a lipschitz attribute.

The whole purpose of this PR is to pass non jitted classes to GeneralizedLinearEstimator. If you must compile them before, it means something is weird, right?

Reply:
Ok, so this is a non standard behavior. Imagine the person reading this piece of code in 2 years: usually we don't need to compile, but here we do. It's not super obvious why. So we should make their life easier by adding a comment, cf my proposition above